### PR TITLE
refactor converter method to seperate class

### PR DESCRIPTION
**Pull request by anetteOlli**

### CODE REVIEW

File: create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusConverter.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+import no.unit.nva.model.EntityDescription;
+import no.unit.nva.model.Identity;
+import no.unit.nva.model.Reference;
+import nva.commons.core.JacocoGenerated;
+import nva.commons.core.paths.UriWrapper;
+
+public class ScopusConverter {
+
+    @JacocoGenerated
+    public ScopusConverter() {
```
Comment (brinxmat) [inline create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusConverter.java:31]: This class is a typical utility class; we aren't instantiating objects, we're simply utilising the methods. Make the entry method static, follow the rabbit down the hole (fortunately your IDE solves this for you).

At the same time, utility classes tend to indicate that something isn't entirely right in object-oriented code (google this and tell me if you think it is pedantic). You might therefore consider renaming the class ScopusPublicationRequest, implementing a constructor that takes a single DocTp as argument, and implementing a public toCreatePublicationRequest method.

That said, I also feel that CreatePublicationRequest is actually an interface and that we are implementing it here.

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusConverter.java (hunk 1 above)
Comment (axthosarouris) [reply create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusConverter.java:31]: My suggestion is basically the same as Rurik's suggestion :P

Comment (anetteOlli) [reply create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusConverter.java]: Fixed to protected class with DocTp in constructor.

I can see that the original implementation is not great object orientation. But having code that conforms to PURE principles is also valuable.

Comment (brinxmat) [reply create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusConverter.java]: By PURE principles, do you mean referential transparency?

Comment (anetteOlli) [reply create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusConverter.java]: https://en.wikipedia.org/wiki/Pure_function

**Review by brinxmat — APPROVED** [review]

**Review by gregorgabriel — APPROVED** [review]

**Pull request merged**